Sources/Confidence/DebugLogger.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+import OSLog
+
+internal protocol DebugLogger {
+    func logEvent(event: ConfidenceEvent, details: String)
+    func logMessage(message: String, isWarning: Bool)
+    func logFlags(flag: String)
+    func logContext(context: ConfidenceStruct)
+}
+
+private extension Logger {
+    private static var subsystem = Bundle.main.bundleIdentifier
+
+    static let confidence = Logger(subsystem: subsystem ?? "", category: "confidence")
+}
+
+internal class DebugLoggerImpl: DebugLogger {
+    func logMessage(message: String, isWarning: Bool) {
+        if !isWarning {
+            Logger.confidence.debug("\(message)")
+        } else {
+            Logger.confidence.warning("\(message)")
+        }
+    }
+
+    func logEvent(event: ConfidenceEvent, details: String) {
+        Logger.confidence.debug("\(details) \(event.name) \(event.payload) \(event.eventTime)")
+    }
+
+    func logFlags(flag: String) {
+        Logger.confidence.debug("\(flag)")
+    }
+
+    func logContext(context: ConfidenceStruct) {
+        Logger.confidence.debug("\(context)")
+    }
```

Comment:
This is gonna be so spammy... Maybe it would be enough to just log something like `context was updated` but omit the actual context data?
Or we introduce an even more verbose level where this is logged.

Reply:
Hm, I am not sure it's possible with OSLog, different levels of logs 🤔 I still need to test out how exactly the logs work, but it's not quite like on Android from what I read.
I don't mind having "context updated" though and we can always add more info in the future iterations